specification/logic/resource-manager/Microsoft.Logic/stable/2019-05-01/logic.json

```diff
@@ -5114,459 +5125,2029 @@
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "integrationServiceEnvironmentName",
+            "description": "The integration service environment name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "default": {
+            "description": "Logic error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+	  "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/operationStatuses/{operationId}": {
+      "get": {
+        "tags": [
+          "integrationServiceEnvironmentOperationStatus"
+        ],
+        "operationId": "IntegrationServiceEnvironment_OperationStatus",
+        "description": "Gets an integration service environment operation status.",
+        "x-ms-examples": {
+          "Gets an integration service environment operation status": {
+            "$ref": "./examples/IntegrationServiceEnvironment_OperationStatus.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "integrationServiceEnvironmentName",
+            "description": "The integration service environment name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "operationId",
+            "description": "The operation id.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AzureAsyncOperationResource"
+            }
+          },
+          "default": {
+            "description": "Logic error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },     
+  	"/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Logic/integrationServiceEnvironments/{integrationServiceEnvironmentName}/health/network": {
```

Comment:
Incorrect segments as noted in previous reviews. RBAC will not work. Please fix.

Reply:
The API confirms with REST standards. Today we just have one health component which is network so we are returning network health here.
As far as RBAC is concerned, anybody who has GET permissions on Integration Service Environment should be able to get the network health. In this case RBAC does not need to be on the health but on the ISE.

Reply:
Customer can have their own custom RBAC. but anyway, since network is an instance of health resource type, as per my other comment, it can be modeled as a parameter which is an enum where the allowed value is "network". In future if you want to support more, you can add to the enum.

Reply:
As discussed closing this.